blocks/rich-text/index.js

```diff
@@ -796,6 +799,19 @@ export class RichText extends Component {
 		} ) );
 	}
 
+	/**
+	 * Calling onSplit means we need to abort the change done by TinyMCE.
+	 * we need to call updateContent to restore the initial content before calling onSplit.
+	 *
+	 * @param {Array}  before content before the split position
+	 * @param {Array}  after  content after the split position
+	 * @param {?Array} blocks blocks to insert at the split position
+	 */
+	restoreContentAndSplit( before, after, blocks ) {
+		this.updateContent();
+		this.props.onSplit( before, after, ...blocks );
```

Comment:
Hm, now we have (1) `this.props.onSplit` checks in several places and (2) none here.